### PR TITLE
MINOR: Add explanation for disabling forwarding from value transformers

### PR DESCRIPTION
**Pull request by astubbs**
Improve code to be self documenting, in cases where forwarding of messages is disabled due to the type of processor being used,

### CODE REVIEW

**Comment by mjsax** [comment]
Retest this please.

**Review by mjsax — APPROVED** [review]
LGTM.

**Comment by mjsax** [comment]
@astubbs -- build failed with SpotBug error

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/ForwardingDisabledProcessorContext.java [hunk 1]
```diff
@@ -40,6 +40,10 @@
 public final class ForwardingDisabledProcessorContext implements ProcessorContext {
     private final ProcessorContext delegate;
 
+    private String explanation = "ProcessorContext#forward() is not supported from this context, "
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/ForwardingDisabledProcessorContext.java]: I think this should be `final` and maybe also `static` ?

**Comment by astubbs** [comment]
Retest this please.

**Comment by bbejeck** [comment]
retest this please

**Comment by bbejeck** [comment]
Retest this please.

**Comment by mjsax** [comment]
Checkstyle error:
```
> Task :streams:checkstyleMain
07:47:53 [ant:checkstyle] [ERROR] /home/jenkins/jenkins-slave/workspace/kafka-pr-jdk14-scala2.13/streams/src/main/java/org/apache/kafka/streams/processor/internals/ForwardingDisabledProcessorContext.java:43:33: Name 'explanation' must match pattern '(^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$)|(^log$)'. [ConstantName]
```

**Comment by astubbs** [comment]
> Checkstyle error:

ah dag nammit, sorry guys

**Comment by mjsax** [comment]
Retest this please.

**Comment by mjsax** [comment]
Thanks for the PR @astubbs!

Merged to `trunk` and cherry-picked to `2.6`.

**Pull request merged**